runtime/binding-kafka/src/main/java/io/aklivity/zilla/runtime/binding/kafka/internal/validator/StringValidator.java

```diff
@@ -15,18 +15,20 @@
  */
 package io.aklivity.zilla.runtime.binding.kafka.internal.validator;
 
+import java.util.function.Predicate;
+
 import org.agrona.DirectBuffer;
 
 import io.aklivity.zilla.runtime.binding.kafka.internal.validator.config.StringValidatorConfig;
 
 public final class StringValidator implements Validator
 {
-    private String encoding;
+    private Predicate<byte[]> predicate;
 
     public StringValidator(
         StringValidatorConfig config)
     {
-        this.encoding = config.encoding;
+        this.predicate = config.encoding.equals("utf_8") ? this::isValidUTF8 : bytes -> false;
```

Comment:
If we only currently support utf8 charset, then please make sure the schema validation only accepts utf8 as a valid encoding, otherwise it will look like we support other encodings, but the implementation will always reject non-utf8.